src/statistics/Statistics.ts

```diff
@@ -0,0 +1,256 @@
+
+            this.fillMetrics(stats, ownerConnection);
+
+            this.addNearCacheStats(stats);
+
+            this.sendStats(stats.join(''), ownerConnection);
+        }, 0, periodSeconds * 1000);
+    }
+
+    sendStats(newStats: string, ownerConnection: ClientConnection) {
+        const request = ClientStatisticsCodec.encodeRequest(newStats);
+
+        this.client.getInvocationService().invokeOnTarget(request, ownerConnection.getAddress()).catch((err) => {
+            // suppress exception, do not print too many messages
+            this.logger.trace('Statistics', 'Could not send stats ', err);
+        });
+    }
+
+    /**
+     * @return the owner connection to the server for the client only if the server supports the client statistics feature
+     */
+    private getOwnerConnection(): ClientConnection {
+        const connection = this.client.getClusterService().getOwnerConnection();
+        if (connection == null) {
+            return null;
+        }
+
+        const ownerConnectionAddress: Address = connection.getAddress();
+        const serverVersion: number = connection.getConnectedServerVersion();
+        if (serverVersion < Statistics.FEATURE_SUPPORTED_SINCE_VERSION) {
+
+            // do not print too many logs if connected to an old version server
+            if (this.ownerAddress == null || !ownerConnectionAddress.equals(this.ownerAddress)) {
+                this.logger.trace('Statistics', 'Client statistics can not be sent to server '
+                    + ownerConnectionAddress + ' since, connected '
+                    + 'owner server version is less than the minimum supported server version ' +
+                    Statistics.FEATURE_SUPPORTED_SINCE_VERSION_STRING);
+
+            }
+            // cache the last connected server address for decreasing the log prints
+            this.ownerAddress = ownerConnectionAddress;
+            return null;
+        }
+        return connection;
+    }
+
+    private registerMetrics() {
+        this.registerGauge('os.committedVirtualMemorySize', () => Statistics.EMPTY_STAT_VALUE);
+        this.registerGauge('os.freePhysicalMemorySize', () => os.freemem());
+        this.registerGauge('os.freeSwapSpaceSize', () => Statistics.EMPTY_STAT_VALUE);
+        this.registerGauge('os.maxFileDescriptorCount', () => Statistics.EMPTY_STAT_VALUE);
+        this.registerGauge('os.openFileDescriptorCount', () => Statistics.EMPTY_STAT_VALUE);
+        this.registerGauge('os.processCpuTime', () => Statistics.EMPTY_STAT_VALUE);
+        this.registerGauge('os.systemLoadAverage', () => '[' + os.loadavg().toString() + ']');
+        this.registerGauge('os.totalPhysicalMemorySize', () => os.totalmem());
+        this.registerGauge('os.totalSwapSpaceSize', () => Statistics.EMPTY_STAT_VALUE);
+        this.registerGauge('runtime.availableProcessors', () => os.cpus().length);
+        this.registerGauge('runtime.freeMemory', () => Statistics.EMPTY_STAT_VALUE);
```

Comment:
we may be able to retrieve some of these information from V8 module: https://nodejs.org/api/v8.html . I am just guessing. Have you checked there?